DisCatSharp.ApplicationCommands/Workers/RegistrationWorker.cs

```diff
@@ -0,0 +1,400 @@
+                return null;
+            }
+            else if (updateList == null)
+            {
+                return null;
+            }
+            else
+            {
+                var discord = ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value;
+                if (discord != null)
+                {
+                    Dictionary<ulong, DiscordApplicationCommand> UpdateCommands = new();
+                    foreach (var cmd in updateList)
+                    {
+                        if (discord.Any(d => d.Name == cmd.Name))
+                        {
+                            UpdateCommands.Add(discord.Where(d => d.Name == cmd.Name).First().Id, cmd);
+                        }
+                    }
+                    return UpdateCommands;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+
+        private static List<ulong> BuildGlobalDeleteList(List<DiscordApplicationCommand> updateList = null)
+        {
+            if (ApplicationCommandsExtension._globalDiscordCommands == null || !ApplicationCommandsExtension._globalDiscordCommands.Any())
+            {
+                return null;
+            }
+            else if (updateList == null)
+            {
+                var discord = ApplicationCommandsExtension._globalDiscordCommands;
+
+                if (discord != null)
+                {
+                    List<ulong> InvalidCommandIds = new();
+                    foreach (var cmd in discord)
+                    {
+                        InvalidCommandIds.Add(cmd.Id);
+                    }
+
+                    return InvalidCommandIds;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            else
+            {
+                var discord = ApplicationCommandsExtension._globalDiscordCommands;
+
+                if (discord != null)
+                {
+                    List<ulong> InvalidCommandIds = new();
+                    foreach (var cmd in discord)
+                    {
+                        if (!updateList.Any(ul => ul.Name == cmd.Name))
+                            InvalidCommandIds.Add(cmd.Id);
+                    }
+
+                    return InvalidCommandIds;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+
+        private static List<DiscordApplicationCommand> BuildGlobalCreateList(List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._globalDiscordCommands == null || !ApplicationCommandsExtension._globalDiscordCommands.Any())
+            {
+                return updateList;
+            }
+            else if (updateList == null)
+            {
+                return null;
+            }
+            else
+            {
```

Comment:
Get rid of the surrounding else block . It is assumed if you reach this point the exit case wasn't met